### DISCUSSION TITLE
BG requirement

### DISCUSSION THREAD

**Discussion by venkatramanm** [beckn/protocol-specifications#301] [Open Commerce Networks]
@core-wg-admin @pramodkvarma @BLR-0118 

1. I can understand why bap may need to fire search on the gateway. (possibly for fair selection of bpp) But why are participants having to send on_search via gateway. This just introduces an unnecessary hop with no benefit. Am I missing some thing here? ONDC currently is mandating sending on_search via bg. Is there any other reason to this. ? 

2. BTW, why even worry about search being sent via gw, I think the fairness logic is also flawed. If some companies wants to form cartel, nothing can be done any way. For that matter, they can do even without ondc. For e.g. If amazon wants to come to ondc as bap and bpp, they may choose to fire to amazon bpp first before firing to rest of bpps. Why should we prevent this? May be we just let market stabilize itself.  But in any case on_search via bg seems wierd to me. 

3. I also want to suggest that if bap sends search via bg, bg should return expected number of bpps = number of ack requests obtained from bpp. That way, baps don't wait till ttl but stop as soon as all bpps expected to respond have responded. 

4. Number of BPPS being broadcasted, may need better filter than city. There may be too many bpps and searches could become too network heavy. 

May be we pick all this up in next core working group call. 

Venky

**Comment by ravi-prakash-v** [comment]
@venkatramanm, going forward lets keep one discussion thread per topic. 

In the CWG meeting held on 16th, #302  Pt no 1 and 3 were addressed. 
In the CWG meeting held on 19th, #306 Pt no 4 was also adressed.

Only Pt no 2 is left to be closed.
